bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/handler/SerialBridgeThingHandler.java

```diff
@@ -31,31 +37,69 @@
 @NonNullByDefault
 public class SerialBridgeThingHandler extends KNXBridgeBaseThingHandler {
 
-    private final SerialClient client;
+    private @Nullable SerialClient client = null;
+    private @Nullable Future<?> initJob = null;
+
+    private final Logger logger = LoggerFactory.getLogger(SerialBridgeThingHandler.class);
 
     public SerialBridgeThingHandler(Bridge bridge) {
         super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        // create new instance using current configuration settings;
+        // when a parameter change is done from UI, dispose() and initialize() are called
         SerialBridgeConfiguration config = getConfigAs(SerialBridgeConfiguration.class);
         client = new SerialClient(config.getAutoReconnectPeriod(), thing.getUID(),
                 config.getResponseTimeout().intValue(), config.getReadingPause().intValue(),
                 config.getReadRetriesLimit().intValue(), getScheduler(), config.getSerialPort(), config.useCemi(),
```

Comment:
Could you also fix the class `BridgeConfiguration` to have `readingPause`, `readRetriesLimit` and `responseTimeout` defined with type `int` and default value set respectively to 50, 3 and 10 to be inline with the thing definitions (ip and serial) ?

Reply:
Allright, lets do some cleanup here, though it makes the patch harder to read. But agreed, I see no reason to use ```BigDecimal``` when the xml description uses ```int``` as well.
For the default values: I do not see a reason to set something in the code, 0 is fine.

Reply:
> For the default values: I do not see a reason to set something in the code, 0 is fine.

The good practice is to align the default in class and the default in XML parameter definition.

Reply:
Hope this does not break anything. For ```autoReconnectPeriod``` I have to choose, 0 (i.e. "never") from ```serial.xml``` vs 60s from ```ip.xml```.

Reply:
@lolodomo I think I want to stick to the values were before. To make is sound, I would need to bring in the defaults in ```BridgeConfiguration.java```, ```IPBridgeConfiguration.java```, ```SerialBridgeConfiguration.java``` and ```DeviceConfig.java```. I do not want to break something shortly before the release. Let us do this in a later patch.

Reply:
OK